### ISSUE TITLE
Incorrect "Unnecessary semicolon" for assignment of lambda to a decorated member

### ISSUE THREAD

**Issue by sebek64** [palantir/tslint#4024]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.7
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export class A<T> {
  @Input()
  public x: (y: T, z: number) => boolean = () => false;
}
```

with `tslint.json` configuration:

recommended (i.e., semicolon rule enabled).

#### Actual behavior

Produces error

ERROR: x.ts[3, 55]: Unnecessary semicolon

#### Expected behavior

Should not produce this error. In particular, the behavior changes when input decorator is removed.

**Comment by JoshuaKGoldberg** [comment]
@sebek64 the `semicolon` rule has a pair of config options:

* `"ignore-bound-class-methods"` skips checking semicolons at the end of bound class methods.
* `"strict-bound-class-methods"` disables any special handling of bound class methods and treats them as any.

Those should address this use case. 🎉
